internal/provider/resource_cloudflare_dlp_profile.go

```diff
@@ -0,0 +1,217 @@
+			newDLPProfile.Entries = append(newDLPProfile.Entries, dlpEntryToAPI(newDLPProfile.Type, entry.(map[string]interface{})))
+		}
+	}
+
+	dlpProfiles, err := client.CreateDLPProfiles(ctx, identifier, cloudflare.CreateDLPProfilesParams{
+		Profiles: []cloudflare.DLPProfile{newDLPProfile},
+		Type:     newDLPProfile.Type,
+	})
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error creating DLP Profile for name %s: %w", newDLPProfile.Name, err))
+	}
+	if len(dlpProfiles) == 0 {
+		return diag.FromErr(fmt.Errorf("error creating DLP Profile for name %s: no profile in response", newDLPProfile.Name))
+	}
+
+	d.SetId(dlpProfiles[0].ID)
+	return resourceCloudflareDLPProfileRead(ctx, d, meta)
+}
+
+func resourceCloudflareDLPProfileUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+
+	updatedDLPProfile := cloudflare.DLPProfile{
+		ID:   d.Id(),
+		Name: d.Get("name").(string),
+		Type: d.Get("type").(string),
+	}
+	updatedDLPProfile.Description, _ = d.Get("description").(string)
+	if entries, ok := d.GetOk("entry"); ok {
+		for _, entry := range entries.(*schema.Set).List() {
+			updatedDLPProfile.Entries = append(updatedDLPProfile.Entries, dlpEntryToAPI(updatedDLPProfile.Type, entry.(map[string]interface{})))
+		}
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("Updating Cloudflare DLP Profile from struct: %+v", updatedDLPProfile))
+
+	identifier := cloudflare.AccountIdentifier(d.Get("account_id").(string))
+	dlpProfile, err := client.UpdateDLPProfile(ctx, identifier, cloudflare.UpdateDLPProfileParams{
+		ProfileID: updatedDLPProfile.ID,
+		Profile:   updatedDLPProfile,
+		Type:      updatedDLPProfile.Type,
+	})
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error updating DLP profile for ID %q: %w", d.Id(), err))
+	}
+	if dlpProfile.ID == "" {
+		return diag.FromErr(fmt.Errorf("failed to find DLP Profile ID in update response; resource was empty"))
+	}
+
+	return resourceCloudflareDLPProfileRead(ctx, d, meta)
+}
+
+func resourceCloudflareDLPProfileDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	tflog.Debug(ctx, fmt.Sprintf("Deleting Cloudflare DLP Profile using ID: %s", d.Id()))
+
+	profileType, _ := d.Get("type").(string)
+	if profileType != DLPProfileTypeCustom {
+		return diag.FromErr(fmt.Errorf("error deleting DLP Profile: can only delete custom profiles"))
+	}
+	identifier := cloudflare.AccountIdentifier(d.Get("account_id").(string))
+	if err := client.DeleteDLPProfile(ctx, identifier, d.Id()); err != nil {
+		return diag.FromErr(fmt.Errorf("error deleting DLP Profile for ID %q: %w", d.Id(), err))
+	}
+
+	resourceCloudflareDLPProfileRead(ctx, d, meta)
+	return nil
+}
+
+func resourceCloudflareDLPProfileImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	attributes := strings.SplitN(d.Id(), "/", 3)
```

Comment:
i think you can do `accountID/dlpProfileID` for the import string instead. unless you're planning on supporting this at the zone level too?